### PR TITLE
Json work

### PR DESCRIPTION
**Pull request by GLTaylor**

### CODE REVIEW

File: Artroom1/AppDelegate.swift [hunk 1]
```diff
@@ -14,10 +14,45 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+        arrayOfArtworks = loadJson("data")!
         
         return true
     }
+      
+        
+        
+        
+        
+//
```
Comment (mAu888) [inline Artroom1/AppDelegate.swift]: I think you can remove this code here 😉

File: Artroom1/ArtPieceDB.swift [hunk 2]
```diff
@@ -8,6 +8,6 @@
...
...icholls", image: "melbourne", attributes: ArtAttributes(mood: .meh, interest: .tech))), (Artwork(title: "Isle Saint-Michel - Elger Esser", image: "Isle Saint-Michel - Elger Esser", attributes: ArtAttributes(mood: .meh, interest: .nature))), (Artwork(title: "Studies of Hands - Alexander Maw", image: "Studies of Hands - Alexander Maw", attributes: ArtAttributes(mood: .meh, interest: .humanity))), (Artwork(title: "Poveglia - Elger Esser", image: "Poveglia - Elger Esser", attributes: ArtAttributes(mood: .meh, interest: .nature))), (Artwork(title: "Drifting - Anthony Goicolea", image: "Drifting - Anthony Goicolea", attributes: ArtAttributes(mood: .meh, interest: .nature))), (Artwork(title: "My parents with their grandson, Ross - Annie Leibovitz", image: "My parents with their grandson, Ross - Annie Leibovitz", attributes: ArtAttributes(mood: .joyful, interest: .humanity))), (Artwork(title: "Class Picture - Anthony Goicolea", image: "Class Picture - Anthony Goicolea", attributes: ArtAttributes(mood: .joyful, interest: .humanity))), (Artwork(title: "By Lantern Light - Erin Nicholls", image: "By Lantern Light - Erin Nicholls", attributes: ArtAttributes(mood: .joyful, interest: .humanity)))
```
Comment (mAu888) [inline Artroom1/ArtPieceDB.swift]: I think you can also get rid of the old data here. Everything is now in your JSON 🙂

File: Artroom1/AppDelegate.swift [hunk 3]
```diff
@@ -14,10 +14,45 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-
+      
+        func loadJson(_ fileName: String) -> [Artwork]? {
```
Comment (mAu888) [inline Artroom1/AppDelegate.swift:18]: Remember when we talked about loading the data etc. and I mentioned you could also move the data loading part into it's own type? I think it's a good time to do that now, so your AppDelegate (this file here) will not get cluttered. In programming there is something called the Single Responsibility Principle. What this means is that any type in your application (class, function etc.) should be focussed on exactly one responsibility.
In this case, as we're in the scope of the `AppDelegate`, loading of data is probably not the responsibility of the delegate. So we could move the `loadJson` function into it's own type (e.g. `DataLoader`). You could then create an instance of the data loader here and call `dataLoader.loadJson()`.

Comment (mAu888) [reply Artroom1/AppDelegate.swift:18]: Another thing I noticed, you are passing a `fileName`, but you are actually not using it in `Bundle.main.path(forResource: ...)` 🙂

Comment (mAu888) [reply Artroom1/AppDelegate.swift:18]: You can also already read up on Singletons. [Here's a quite good and easy introduction](https://cocoacasts.com/what-is-a-singleton-and-how-to-create-one-in-swift) (at least I thought so while skimming over it).

Comment (GLTaylor) [reply Artroom1/AppDelegate.swift:18]: Coming in the next branch!

**Review by mAu888 — APPROVED** [review]
Nice, this one is good to go in my opinion 🚀

**Pull request merged**